clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/PartitionConsumptionState.java

```diff
@@ -565,6 +573,10 @@ public void setTransientRecord(
       int valueLen,
       int valueSchemaId,
       GenericRecord replicationMetadataRecord) {
+    if (this.threadSafeMode) {
+      // NoOp
```

Comment:
I think the problem then becomes how do we shrink the TR then.  Right now theres an implicit TR max size that's the size of the number of requests in flight between consumption and commit to rocksdb (we evict it once it's gone through the drainer).  In this new scheme I'm not sure when we'd evict.  One way to do that would be to swap the TR cache for a caffeing cache thats size/time bound.  But I'm trying to keep this first patch as small as possible.

Reply:
Actually, one of the goals of this refactoring is to get rid of the `TRC`, since that is going to yield an efficiency improvement, and reduce overall complexity. IOW... if there is a reason why we would need to keep the TRC, then I think we need to scrutinize this thoroughly, and I wonder in that case if the refactoring is worth it at all...